### PR TITLE
Initial import

### PR DESCRIPTION
**Pull request by jponge**

### CODE REVIEW

File: dependency-reduced-pom.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tsegismont) [inline dependency-reduced-pom.xml]: Why is this filed in source control?

**Comment by jponge** [comment]
A mistake :-)

File: README.adoc [hunk 2]
```diff
@@ -1,180 +1,192 @@
...
-For instance a Maven `pom.xml` is co-located with `README.adoc` at the root of the repository.
+Vert.x is a resource-efficient toolkit for building all kinds of modern distributed applications, and OpenJ9 is a resource-efficient runtime that is well-suited for virtualized and containerized deployments.
 
-Projects should preferably have both Maven and Gradle build descriptors.
-An author who doesn't have knowledge of either of the tools can always ask for contributions.
+== What you will build and run
 
-== Structure of a how-to
+* You will build a simple micro-service that computes the sum of 2 numbers through a HTTP JSON endpoint.
+* We will check the options for improving startup time with OpenJ9.
```
Comment (tsegismont) [inline README.adoc]:
```suggestion
* We will look at the options for improving startup time with OpenJ9.
```

File: README.adoc [hunk 3]
```diff
@@ -1,180 +1,192 @@
...
+* We will discuss how to improve the startup time of Docker containers and how to tune OpenJ9 in that environment.
 
-A how-to is not a guide or a mini-book: it must be as concise as possible.
+== What you need
 
-In practice this means that each how-to must focus on a very narrow topic.
-It also means that the text must not deviate from the topic, and that the chosen example must be as simple as possible.
+- A text editor or IDE
+- Java 8 higher
+- OpenJ9 (we recommend a build from https://adoptopenjdk.net/[AdoptOpenJDK])
```
Comment (tsegismont) [inline README.adoc:22]:
```suggestion
- OpenJ9 (we recommend a build from https://adoptopenjdk.net/?jvmVariant=openj9[AdoptOpenJDK])
```

File: README.adoc [hunk 4]
```diff
@@ -1,180 +1,192 @@
...
-- Always show the various options to run the application.
+We can run the service:
 
-=== Summary
+----
+$ ./gradlew run
+----
 
-- Briefly recapitulate what was covered and learned in the how-to.
+We can also build a JAR archive will all dependencies bundled, then execute it:
```
Comment (tsegismont) [inline README.adoc]: Move the `shadowJar` instruction down before you instruct to run the JAR with `java -jar`

File: README.adoc [hunk 5]
```diff
@@ -1,180 +1,192 @@
...
-The common theme depends on the following attributes to be set right below the title header:
+The micro-service reports the startup time by measuring the time between the `main` method entry, and the callback notification when the HTTP server has started.
 
-[source,text]
-----
-= My awesome how-to
-:page-permalink: /
-:page-github: vertx-howtos/my-howto
-----
+We can do a few runs of `java -jar build/libs/openj9-howto-all.jar` and pick the best time.
```
Comment (tsegismont) [inline README.adoc:89]: Clarify here if you're talking about running with Hotspot or J9 without config. If you're talking about running J9 without config, can you add a note about configuring the `PATH` before.

File: README.adoc [hunk 6]
```diff
@@ -1,180 +1,192 @@
...
+Let us simulate 2000 users with a 50 new users per second hatch rate.
+This gives us about 600 requests per second.
 
-----
-$ bundle install
-----
+=== Measuring RSS
 
-From time to time you may also update dependencies:
+The https://quarkus.io/guides/performance-measure[Quarkus team has a good guide on measuring RSS].
```
Comment (tsegismont) [inline README.adoc:134]: So the link to the `lengthy` article is nice :wink:

But can we get a TL;DR here to take our own measurements?

File: README.adoc [hunk 7]
```diff
@@ -1,180 +1,192 @@
...
+----
+$ docker build . -t openj9-app
+----
 
-=== Code style
+We can then create containers from the image:
 
-There is a `.editorconfig` file in this template repository with the 2-space indentation style that we prefer in Vert.x.
+----
+$ docker run -it --rm -v /tmp/_cache:/app/_cache -p 8080:8080 openj9-app
```
Comment (tsegismont) [inline README.adoc:202]: On some platforms with `cgroups` active the container cannot access the shared classes directory. They will get an error like:

```
JVMSHRC226E Error opening shared class cache file 
JVMSHRC336E Port layer error code = -102
JVMSHRC337E Platform error message: Permission denied
JVMJ9VM015W Initialization error for library j9shr29(11): JVMJ9VM009E J9VMDllMain failed
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

Can you add a `TIP` about this and say they can relax the check with `-v /tmp/_cache:/app/_cache:z`

**Review by tsegismont — CHANGES_REQUESTED** [review]
@jponge great work! A few comments/concerns

File: README.adoc (hunk 3 above)
Comment (jponge) [reply README.adoc:22]: I'd rather leave people pick the OpenJDK base version that they want.

File: README.adoc (hunk 7 above)
Comment (jponge) [reply README.adoc:202]: 👍

**Review by pmlopes — APPROVED** [review]

**Comment by jponge** [comment]
@jponge @tsegismont changes have been made

File: README.adoc (hunk 3 above)
Comment (tsegismont) [reply README.adoc:22]: Right. So I updated the `url` so that it still forces the `openj9` variant.

**Review by tsegismont — COMMENTED** [review]
LGTM

**Pull request merged**